server/common/common-service/src/main/java/io/holoinsight/server/common/service/flyway/FlywayService.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2022 Holoinsight Project Authors. Licensed under Apache-2.0.
+ */
+package io.holoinsight.server.common.service.flyway;
+
+import java.util.List;
+
+import javax.annotation.PostConstruct;
+import javax.sql.DataSource;
+
+import io.holoinsight.server.common.dao.mapper.FlywaySchemaHistoryDOMapper;
+import lombok.extern.slf4j.Slf4j;
+import org.flywaydb.core.Flyway;
+import org.flywaydb.core.api.configuration.FluentConfiguration;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.config.BeanPostProcessor;
+import org.springframework.util.CollectionUtils;
+
+/**
+ * @author jiwliu
+ * @date 2023/3/1
+ */
+
+@Slf4j
+public class FlywayService implements BeanPostProcessor {
+
+  public static final String REPAIR = "repair";
+  public static final String MIGRATE = "migrate";
+  public static final String SKIP_ALL = "skipAll";
+  public static final String SKIP_LATEST = "skipLatest";
+  public static final String LIST = "list";
+  private final Flyway flyway;
+
+  @Autowired
+  private FlywaySchemaHistoryDOMapper flywaySchemaHistoryDOMapper;
+
+  public FlywayService(FlywayProperties properties, DataSource dataSource) {
+    FluentConfiguration config = Flyway.configure().dataSource(dataSource);
+    fillConfig(config, properties);
+    this.flyway = config.load();
+  }
+
+  @PostConstruct
+  public void migrate() {
+    try {
+      flyway.migrate();
+    } catch (Exception e) {
+      log.error("[flywaydb] call migrate error", e);
```

Comment:
Should throw an exception here ?
I think database schema initialization failure is a fatal error. It should block the bootstrap.